Examples/UIExplorer/AsyncStorageExample.js

```diff
@@ -29,15 +29,16 @@ var COLORS = ['red', 'orange', 'yellow', 'green', 'blue'];
 
 var BasicStorageExample = React.createClass({
   componentDidMount() {
-    AsyncStorage.getItem(STORAGE_KEY, (error, value) => {
-      if (error) {
-        this._appendMessage('AsyncStorage error: ' + error.message);
-      } else if (value !== null) {
-        this.setState({selectedValue: value});
-        this._appendMessage('Recovered selection from disk: ' + value);
-      } else {
-        this._appendMessage('Initialized with no selection on disk.');
-      }
+    AsyncStorage.getItem(STORAGE_KEY)
+      .then(value => {
+        if(value !== null){
+          this.setState({selectedValue: value});
+          this._appendMessage('Recovered selection from disk: ' + value);
+        } else {
+          this._appendMessage('Initialized with no selection on disk.');
+        }
+      })
+      .catch((error) => this._appendMessage('AsyncStorage error: ' + error.message));
```

Comment:
I think if you don't call `.done()` at the end of these promise chains, it might swallow other errors - can you add it?